src/writeFile.js

```diff
@@ -23,7 +26,34 @@ export default function writeFile(globalRef, pattern, file) {
         return fs.readFileAsync(file.absoluteFrom)
         .then((content) => {
             if (pattern.transform) {
-                content = pattern.transform(content, file.absoluteFrom);
+                const transform = (content, absoluteFrom) => {
+                    return pattern.transform(content, absoluteFrom);
+                };
+
+                if (pattern.cache) {
+                    const cacheKey = pattern.cacheKey
```

Comment:
Do we really need to ability to customize  the `cacheKey` via `pattern.options` ?

Reply:
@michael-ciniawsky yep, we can use package inside `transform` function which get options, to invalidate cache you should pass this variables as cacheKey